public/containers/ClearCacheModal/ClearCacheModal.tsx

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { CoreStart } from "opensearch-dashboards/public";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+import { ServicesContext } from "../../services";
+import { CoreServicesContext } from "../../components/core_services";
+import { IndexOpBlocksType, SOURCE_PAGE_TYPE } from "../../utils/constants";
+import {
+  getErrorMessage,
+  aliasBlockedPredicate,
+  indexBlockedPredicate,
+  dataStreamBlockedPredicate,
+  filterBlockedItems,
+} from "../../utils/helpers";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+
+interface ClearCacheModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  onClose: () => void;
+  type: string;
+}
+
+export default function ClearCacheModal<T>(props: ClearCacheModalProps<T>) {
+  const { onClose, visible, selectedItems, type } = props;
+  const [hint, setHint] = useState("");
+  const [blockHint, setBlockHint] = useState("");
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+
+  useEffect(() => {
+    const indexBlocksTypes = [
+      IndexOpBlocksType.Closed,
+      IndexOpBlocksType.MetaData,
+      IndexOpBlocksType.ReadOnly,
+      IndexOpBlocksType.ReadOnlyAllowDelete,
+    ];
+    if (!!services && visible) {
+      switch (type) {
+        case SOURCE_PAGE_TYPE.DATA_STREAMS:
+          setHint("Caches will be cleared for the following data streams.");
+          setBlockHint(
+            "Caches will not be cleared for the following data streams because one or more backing indexes are closed or blocked."
+          );
+          filterBlockedItems<DataStream>(services, selectedItems as DataStream[], indexBlocksTypes, dataStreamBlockedPredicate).then(
```

Comment:
We should add .catch logic to handle errors(e.g. no permission) in filterBlockedItems function. I think in this case we can degrade to not filtering blocked items, i.e. set selectedItems as nonBlockedItems.